tools/report-converter/codechecker_report_converter/util.py

```diff
@@ -76,8 +78,16 @@ def trim_path_prefixes(path: str, prefixes: Optional[List[str]]) -> str:
         if not prefix.endswith('/'):
             prefix += '/'
 
-        if path.startswith(prefix) and (not longest_matching_prefix or
-                                        longest_matching_prefix < prefix):
+        regex_str = fnmatch.translate(prefix)
+        if (regex_str[-2:] != '\\Z'):
+            print(regex_str[-2:])
+        assert regex_str[-2:] == '\\Z', \
+               r'fnmatch.translate should leave \\Z at the end of the matcher!'
+
+        prefix_matcher = re.compile(regex_str[:-2])
+
+        if prefix_matcher.match(path) and \
+           (not longest_matching_prefix or longest_matching_prefix < prefix):
```

Comment:
```suggestion
           (not longest_matching_prefix or len(longest_matching_prefix) < len(prefix)):
```
It is interesting that earlier we checked alphabetical order on matching path prefixes. According to the specification we should check their lengths. Would you agree with this fix, or am I missing something?